### PR TITLE
[spec] Address review by eholk

### PR DESCRIPTION
**Pull request by rossberg**
Address #723 and also #781.

@conrad-watt, note the change to the definition of valid host function. Will your proof still go through with the modified definition?

### CODE REVIEW

**Comment by conrad-watt** [comment]
Hi Andreas, there should be no problems with the soundness proof.

To be clear, when _hf_ is not defined, does this mean the host function diverges? Because the reduction to trap would still be enabled, which seems a little imprecise. I can't think of a neat alternative though.

This all changes with threads, anyway.

**Comment by conrad-watt** [comment]
Ah ok, then I think there would be a problem with the proof, since the progress property would be violated if the function is undefined.

**Comment by rossberg** [comment]
Done. Actually, it has to be deterministic, otherwise we would allow divergence even for host functions that are total.

**Comment by rossberg** [comment]
Hm, not without substantial hand-waving (such as the host could extend the definition of store). But that wasn't the case before either. I suppose one could make the function set-valued to model real non-determinism, but it would also require a way to express the case where it can either diverge or deliver a result. I guess that would require making divergence more explicit with an additional bottom result.

**Comment by rossberg** [comment]
Done. Somebody please double-check the definitions carefully.

File: document/core/appendix/properties.rst [hunk 1]
```diff
@@ -714,7 +721,7 @@ the standard soundness theorems hold. [#cite-cpp2018]_
 **Theorem (Preservation).**
 If a :ref:`configuration <syntax-config>` :math:`S;T` is :ref:`valid <valid-config>` with :ref:`result type <syntax-resulttype>` :math:`[t^\ast]` (i.e., :math:`\vdashconfig S;T : [t^\ast]`),
 and steps to :math:`S';T'` (i.e., :math:`S;T \stepto S';T'`),
-then :math:`S';T'` is a valid configuration with the same resulttype (i.e., :math:`\vdashconfig S';T' : [t^\ast]`).
+then :math:`S';T'` is a valid configuration with the same result type (i.e., :math:`\vdashconfig S';T' : [t^\ast]`).
```
[the remark this answers is not in the document]
Comment (rossberg) [reply document/core/appendix/properties.rst]: I usually only linked the first occurrence in a sentence or paragraph.

**Comment by rossberg** [comment]
@conrad-watt, let me know if you think the latest def of host functions looks right to you.

**Comment by conrad-watt** [comment]
LGTM!

**Pull request merged**